src/cloudai/_core/configurator/cloudai_gym.py

```diff
@@ -116,12 +118,21 @@ def step(self, action: Any) -> Tuple[list, float, bool, dict]:
         """
         for key, value in action.items():
             self.update_nested_attr(self.test_run.test.test_definition.cmd_args, key, value)
+
+        constraint_check = getattr(self.test_run.test.test_definition, "constraint_check", None)
```

Comment:
These two lines should become `if not self.test_run.test.test_definition.constraint_check:`.